### ISSUE TITLE
bet2 dependencies issue

### ISSUE THREAD

**Issue by 4rnaudB** [shimming-toolbox/shimming-toolbox#553]
### Bug Description
The tests suddenly stopped working with no code change. The problem seems to be caused by bet2's installation from fsl.
```
echo "Download BET from FSL"
  source ~/shimming-toolbox/python/bin/activate
  conda install -c https://fsl.fmrib.ox.ac.uk/fsldownloads/fslconda/public/ -c conda-forge fsl-bet2
  shell: /usr/bin/bash -e {0}
  env:
    FSLOUTPUTTYPE: NIFTI_GZ
    pythonLocation: /opt/hostedtoolcache/Python/[3](https://github.com/shimming-toolbox/shimming-toolbox/actions/runs/11220855596/job/31189917387#step:14:3).10.15/x64
    LD_LIBRARY_PATH: /opt/hostedtoolcache/Python/3.10.15/x6[4](https://github.com/shimming-toolbox/shimming-toolbox/actions/runs/11220855596/job/31189917387#step:14:4)/lib
Download BET from FSL
Channels:
 - https://fsl.fmrib.ox.ac.uk/fsldownloads/fslconda/public
 - conda-forge
Platform: linux-64
Collecting package metadata (repodata.json): ...working... done
warning  libmamba Added empty dependency for problem type SOLVER_RULE_UPDATE
Solving environment: ...working... failed

LibMambaUnsatisfiableError: Encountered problems while solving:
  - package libxml2-2.12.7-he7c6b[5](https://github.com/shimming-toolbox/shimming-toolbox/actions/runs/11220855596/job/31189917387#step:14:5)8_4 requires icu >=75.1,<76.0a0, but none of the providers can be installed

Could not solve for environment specs
The following packages are incompatible
├─ fsl-bet2 is installable with the potential options
│  ├─ fsl-bet2 [2111.0|2111.1|2111.2] would require
│  │  └─ boost-cpp >=1.74.0,<1.74.1.0a0  with the potential options
│  │     ├─ boost-cpp 1.74.0 would require
│  │     │  └─ icu >=[6](https://github.com/shimming-toolbox/shimming-toolbox/actions/runs/11220855596/job/31189917387#step:14:6)8.1,<69.0a0 , which can be installed;
│  │     ├─ boost-cpp 1.[7](https://github.com/shimming-toolbox/shimming-toolbox/actions/runs/11220855596/job/31189917387#step:14:7)4.0 would require
│  │     │  └─ icu >=69.1,<70.0a0 , which can be installed;
│  │     ├─ boost-cpp [1.74.0|1.[8](https://github.com/shimming-toolbox/shimming-toolbox/actions/runs/11220855596/job/31189917387#step:14:8)1.0] would require
│  │     │  └─ icu >=70.1,<71.0a0 , which can be installed;
│  │     ├─ boost-cpp 1.74.0 would require
│  │     │  ├─ icu >=67.1,<68.0a0 , which can be installed;
│  │     │  └─ zstd >=1.4.5,<1.5.0a0 , which can be installed;
│  │     ├─ boost-cpp 1.74.0 would require
│  │     │  ├─ icu >=68.1,<6[9](https://github.com/shimming-toolbox/shimming-toolbox/actions/runs/11220855596/job/31189917387#step:14:9).0a0 , which can be installed;
│  │     │  └─ zstd >=1.4.5,<1.5.0a0 , which can be installed;
│  │     ├─ boost-cpp 1.74.0 would require
│  │     │  ├─ icu >=68.1,<69.0a0 , which can be installed;
│  │     │  └─ zstd >=1.4.8,<1.5.0a0 , which can be installed;
│  │     └─ boost-cpp 1.74.0 would require
│  │        ├─ icu >=68.1,<69.0a0 , which can be installed;
│  │        └─ zstd >=1.4.9,<1.5.0a0 , which can be installed;
│  ├─ fsl-bet2 2111.2 would require
│  │  └─ boost-cpp >=1.81.0,<1.81.1.0a0  with the potential options
│  │     ├─ boost-cpp [1.74.0|1.81.0], which can be installed (as previously explained);
│  │     └─ boost-cpp 1.81.0 would require
│  │        └─ icu >=72.1,<73.0a0 , which can be installed;
│  └─ fsl-bet2 [2111.3|2111.4|...|2111.8] would require
│     └─ fsl-utils [ |>=2203.2,<2204.0a0 |>=2203.4,<2204.0a0 ] with the potential options
│        ├─ fsl-utils 2203.2 would require
│        │  └─ boost-cpp >=1.74.0,<1.74.1.0a0 , which can be installed (as previously explained);
│        ├─ fsl-utils 2203.2 would require
│        │  └─ boost-cpp >=1.81.0,<1.81.1.0a0 , which can be installed (as previously explained);
│        └─ fsl-utils 2203.4 would require
│           ├─ fsl-armawrap with the potential options
│           │  ├─ fsl-armawrap 0.6.0 would require
│           │  │  └─ boost-cpp >=1.74.0,<1.74.1.0a0 , which can be installed (as previously explained);
│           │  ├─ fsl-armawrap 0.6.0 would require
│           │  │  └─ boost-cpp >=1.81.0,<1.81.1.0a0 , which can be installed (as previously explained);
│           │  └─ fsl-armawrap 0.6.0 would require
│           │     └─ libzlib >=1.2.13,<1.3.0a0 , which can be installed;
│           └─ libzlib >=1.2.13,<1.3.0a0 , which can be installed;
├─ mamba >=1.5.9  is installable with the potential options
│  ├─ mamba [1.5.[10](https://github.com/shimming-toolbox/shimming-toolbox/actions/runs/11220855596/job/31189917387#step:14:11)|1.5.9|2.0.0|2.0.1|2.0.2] would require
│  │  ├─ libmamba [2.0.0 h188e5ea_0|2.0.0 hf72d635_1|2.0.1 hf72d635_0|2.0.2 hf72d635_0], which requires
│  │  │  ├─ libarchive >=3.7.4,<3.8.0a0 , which requires
│  │  │  │  └─ libxml2 >=2.12.7,<3.0a0  but there are no viable options
│  │  │  │     ├─ libxml2 2.12.7 would require
│  │  │  │     │  ├─ icu >=75.1,<76.0a0 , which conflicts with any installable versions previously reported;
│  │  │  │     │  └─ libzlib >=1.3.1,<2.0a0 , which conflicts with any installable versions previously reported;
│  │  │  │     ├─ libxml2 2.12.7 would require
│  │  │  │     │  └─ icu >=73.2,<74.0a0 , which conflicts with any installable versions previously reported;
│  │  │  │     └─ libxml2 2.12.7 would require
│  │  │  │        └─ icu >=75.1,<76.0a0 , which conflicts with any installable versions previously reported;
│  │  │  ├─ libcurl [>=8.10.1,<9.0a0 |>=8.9.1,<9.0a0 ] but there are no viable options
│  │  │  │  ├─ libcurl 8.10.1 would require
│  │  │  │  │  ├─ libzlib >=1.3.1,<2.0a0 , which conflicts with any installable versions previously reported;
│  │  │  │  │  └─ zstd >=1.5.6,<1.6.0a0 , which conflicts with any installable versions previously reported;
│  │  │  │  └─ libcurl [8.10.0|8.9.1] would require
│  │  │  │     └─ libzlib >=1.3.1,<2.0a0 , which conflicts with any installable versions previously reported;
│  │  │  ├─ libsolv [>=0.7.30,<0.7.31.0a0 |>=0.7.30,<0.8.0a0 ], which requires
│  │  │  │  └─ libzlib >=1.3.1,<2.0a0 , which conflicts with any installable versions previously reported;
│  │  │  └─ zstd >=1.5.6,<1.6.0a0 , which conflicts with any installable versions previously reported;
│  │  └─ libmambapy [1.5.10 py310h6639945_1|1.5.10 py310h86cbe3b_0|1.5.9 py310h86cbe3b_0], which requires
│  │     └─ libmamba [1.5.10 h4cc3d14_0|1.5.10 hf72d635_1|1.5.9 h4cc3d14_0], which cannot be installed (as previously explained);
│  ├─ mamba [1.5.10|1.5.9] would require
│  │  └─ python >=3.[11](https://github.com/shimming-toolbox/shimming-toolbox/actions/runs/11220855596/job/31189917387#step:14:12),<3.12.0a0 , which can be installed;
│  ├─ mamba [1.5.10|1.5.9] would require
│  │  └─ python >=3.[12](https://github.com/shimming-toolbox/shimming-toolbox/actions/runs/11220855596/job/31189917387#step:14:13),<3.[13](https://github.com/shimming-toolbox/shimming-toolbox/actions/runs/11220855596/job/31189917387#step:14:14).0a0 , which can be installed;
│  └─ mamba [1.5.10|1.5.9] would require
│     └─ python >=3.9,<3.10.0a0 , which can be installed;
└─ pin-1 is not installable because it requires
   └─ python 3.10.* , which conflicts with any installable versions previously reported.

Pins seem to be involved in the conflict. Currently pinned specs:
 - python 3.10.* (labeled as 'pin-1')
 ```

**Comment by 4rnaudB** [comment]
Tried changing python 3.10* to python 3.11* without success

**Comment by joshuacwnewton** [comment]
- First failure: [Sept 6, 2024](https://github.com/shimming-toolbox/shimming-toolbox/actions/runs/10741244770/job/29791168356)
   - Reading the logs, it seems the incompatible packages are `fsl-bet2` and `mamba`. Both of these packages expect different versions of the subpackages `icu`, `libzlib`, and `zstd`, hence the conflict.
   - Under Python 3.10.x, the dependency solver tries to test the following versions:
      - `fsl-bet2`: `[2111.0, 2111.8]` (see [this line](https://github.com/shimming-toolbox/shimming-toolbox/actions/runs/10741244770/job/29791168356#step:14:51))
      - `mamba`: `1.5.9`
- Last success: [Aug 21, 2024](https://github.com/shimming-toolbox/shimming-toolbox/actions/runs/10497324978/job/29079910168#step:14:11)
   - When checking the log of the successful run, we can see the following packages get installed:
      - `fsl-bet2`: `2111.8`
      - `mamba`: `1.5.8` (listed in the "Package ST installation" section under `libmamba`)

Given that the `fsl-bet2` version hasn't changed since the last passing release (`latest==2111.8`), this is a hint that the bug is possibly due to `mamba==1.5.9`. And in fact, when I check the logs a little more closely, I see that there is a new warning that wasn't present in the passing run:

```diff
Download BET from FSL
Channels:
 - https://fsl.fmrib.ox.ac.uk/fsldownloads/fslconda/public
 - conda-forge
Platform: linux-64
Collecting package metadata (repodata.json): ...working... done
+warning  libmamba Added empty dependency for problem type SOLVER_RULE_UPDATE
Solving environment: ...working...
```

I tried googling this warning, but there are a lot of different reports from many different time periods, so I don't think there is a clear way to interpret this message.

Still, if I look at the ST installer, I see you use the `latest` Mambaforge installer from https://github.com/conda-forge/miniforge/releases/. This is a good clue, as the installer may have auto-updated in the last few months (despite no code changes).

Looking at the release history, I see that there was a release on July 24 ([Release 24.5.0-0](https://github.com/conda-forge/miniforge/releases/tag/24.5.0-0)) and another on September 3 ([Release 24.7.1-0](https://github.com/conda-forge/miniforge/releases/tag/24.7.1-0)). This matches the CI success/failure timeline very closely. So, I would temporarily replace `latest` with the version tag of the last working release (July 24th) [here](https://github.com/shimming-toolbox/shimming-toolbox/blob/f0fae7f27527d78da4179e508d7cd09d95581fc7/installer/install_conda.sh#L42) and see if that fixes the issue. 

If it does fix the issue, then the end result here may be limit `mamba<1.5.9` for now when creating the ST environment, and report to the `fsl-bet2` maintainers that it is incompatible with `mamba==1.5.9` so they can update their pinned dependencies and release e.g. `2111.9`.

**Comment by 4rnaudB** [comment]
Fixed mamba in PR #554. There is a documentation test failing, I don't know if it is related, but I will investigate more. Thanks for the help.
